src/open-id-connect.ts

```diff
@@ -25,8 +25,8 @@ export default class OpenIdConnect {
     this.openIdConnectRouting.configureRouter(routerConfiguration);
   }
 
-  public async login(): Promise<void> {
-    const args: any = {};
+  public async login(redirectRequestRoute: boolean = false): Promise<void> {
+    const args: any = redirectRequestRoute ? {data: this.router.currentInstruction.fragment} : {};
```

Comment:
This is where we could use pass the `redirect_url` instead of `data`. 

    const loginRedirectRoute = decodeURIComponent(/*loginRedirectRouteFromQueryString*/);
    const args = { redirect_url: loginRedirectRoute };
    await this.userManager.signinRedirect(args);

I've put more details about this in my first comment to this PR.